File: Desarrollos AddOn SAP B1/Samples/COM DI/CSharp/10.Message/MainModule.cs
using System.Diagnostics;
using Microsoft.VisualBasic;
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using System.Collections;
using SAPbobsCOM;

//  SAP DI API 2006 SDK Sample
//****************************************************************************
//
//  File:      MainModule.vb
//
//  Copyright (c) SAP
//
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
//****************************************************************************

//****************************************************************************
//
// Description:
// -------------
// 1) Log in to the company
// 2) Display the messages : Inbox , Outbox, SendMessages
// 3) Display how to send a message

//****************************************************************************



namespace FormWindowTemplateVb
{
	sealed class MainModule
	{
		public static SAPbobsCOM.Company oCompany = new SAPbobsCOM.Company();
		public static SAPbobsCOM.CompanyService oCmpSrv;
		public static MessagesService oMessageService;
		
		static public void Main ()
		{
			
			StartupForm frm = new StartupForm();
			
			frm.ShowDialog();
			
		}
		
		//init the service object
		public static void InitService ()
		{
			
			
			//get company service
			oCmpSrv = oCompany.GetCompanyService();
			
			//get msg service
			oMessageService = (SAPbobsCOM.MessagesService) oCmpSrv.GetBusinessService(ServiceTypes.MessagesService);
			
			
		}
		
	}
	
}

File: Desarrollos AddOn SAP B1/Samples/COM DI/VB.NET/10.Messages Service/MainModule.vb
'  SAP DI API 2006 SDK Sample
'****************************************************************************
'
'  File:      MainModule.vb
'
'  Copyright (c) SAP 
'
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
' ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
' THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'
'****************************************************************************

'****************************************************************************
'
' Description:
' -------------
' 1) Log in to the company
' 2) Display the messages : Inbox , Outbox, SendMessages
' 3) Display how to send a message  

'****************************************************************************

Imports SAPbobsCOM


Module MainModule
    Public oCompany As New SAPbobsCOM.Company
    Public oCmpSrv As SAPbobsCOM.CompanyService
    Public oMessageService As MessagesService

    Sub Main()

        Dim frm As New StartupForm

        frm.ShowDialog()

    End Sub

    'init the service object
    Public Sub InitService()


        'get company service
        oCmpSrv = oCompany.GetCompanyService

        'get msg service
        oMessageService = oCmpSrv.GetBusinessService(ServiceTypes.MessagesService)


    End Sub

End Module
